src/main/java/com/avioconsulting/mule/connector/vault/provider/internal/connection/provider/VaultConnectionProvider.java

```diff
@@ -49,23 +51,38 @@
 
   @DisplayName("Secrets Engine Version")
   @Parameter
-  @Optional
+  @Optional(defaultValue = "v1")
   private EngineVersion engineVersion;
 
   @DisplayName("Vault Token")
   @Parameter
   private String vaultToken;
 
   @Parameter
+  @Placement(tab = "Security")
   @Optional
   private TlsContextFactory tlsContextFactory;
 
+  @DisplayName("Response Timeout")
+  @Summary("Maximum time to wait for a response in milliseconds")
+  @Parameter
+  @Placement(tab = "Settings")
+  @Optional(defaultValue = "5000")
+  private Integer responseTimeout;
```

Comment:
See duration parameters at https://docs.mulesoft.com/mule-sdk/1.1/general-coding-rules.  There should be a duration and TimeUnit parameter